File: src/main/java/com/vaadin/demo/component/datetimepicker/DateTimePickerReadonlyAndDisabled.java
package com.vaadin.demo.component.datetimepicker;

import com.vaadin.flow.component.datetimepicker.DateTimePicker;
import com.vaadin.flow.component.formlayout.FormLayout;
import com.vaadin.flow.router.Route;

import java.time.LocalDateTime;

import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("date-time-picker-readonly-and-disabled")
public class DateTimePickerReadonlyAndDisabled extends FormLayout {

    public DateTimePickerReadonlyAndDisabled() {
        setAutoResponsive(true);
        setColumnWidth("20rem");

        // tag::snippet[]
        DateTimePicker readonlyField = new DateTimePicker();
        readonlyField.setReadOnly(true);
        readonlyField.setLabel("Read-only");
        readonlyField.setValue(LocalDateTime.of(2020, 6, 12, 12, 30));

        DateTimePicker disabledField = new DateTimePicker();
        disabledField.setEnabled(false);
        disabledField.setLabel("Disabled");
        // end::snippet[]

        add(readonlyField, disabledField);
    }

    public static class Exporter extends // hidden-source-line
            DemoExporter<DateTimePickerReadonlyAndDisabled> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/datetimepicker/date-time-picker-readonly-and-disabled.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import '@vaadin/vertical-layout';
import '@vaadin/date-time-picker';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('date-time-picker-readonly-and-disabled')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <vaadin-vertical-layout>
        <!-- tag::snippet[] -->
        <vaadin-date-time-picker readonly label="Read-only" value="2020-06-12T12:30">
        </vaadin-date-time-picker>

        <vaadin-date-time-picker disabled label="Disabled"></vaadin-date-time-picker>
        <!-- end::snippet[] -->
      </vaadin-vertical-layout>
    `;
  }
}
